src/async_fuse/memfs/kv_engine.rs

```diff
@@ -276,34 +292,35 @@ impl MetaTxn for EtcdTxn {
         if self.version_map.is_empty() && self.buffer.is_empty() {
             return Ok(true);
         }
-        let mut req: etcd_client::EtcdTxnRequest = etcd_client::EtcdTxnRequest::new();
-        // add the version check
-        for (key, value) in &self.version_map {
-            let version = *value;
-            req = req.when_version(
-                etcd_client::KeyRange::key(key.clone()),
-                TxnCmp::Equal,
-                version,
-            );
-        }
-        // add the write operations
-        for (key, value) in &self.buffer {
-            if let Some(ref value) = *value {
-                let put_req = etcd_client::EtcdPutRequest::new(key.clone(), value.clone());
-                req = req.and_then(put_req);
-            } else {
-                let delete_req =
-                    etcd_client::EtcdDeleteRequest::new(etcd_client::KeyRange::key(key.clone()));
-                req = req.and_then(delete_req);
-            }
-        }
+
         let resp = self
             .client
-            .kv()
-            .txn(req)
+            .txn(
+                Txn::new()
+                    .when(
+                        self.version_map
+                            .iter()
+                            .map(|(key, version)| {
+                                Compare::version(key.clone(), CompareOp::Equal, *version)
+                            })
+                            .collect::<Vec<Compare>>(),
+                    )
+                    .and_then(
+                        self.buffer
+                            .iter()
+                            .map(|(key, value)| {
+                                if let Some(ref value) = *value {
+                                    TxnOp::put(key.clone(), value.clone(), None)
+                                } else {
+                                    TxnOp::delete(key.clone(), None)
+                                }
+                            })
+                            .collect::<Vec<TxnOp>>(),
+                    ),
+            )
```

Comment:
:smiley: Using such a functional programming style is commendable. In some cases, this style can bring higher performance than using a for loop. The reason is that functional programming style compilers can obtain more information, which may enable them to make some optimizations(https://github.com/risinglightdb/risinglight/issues/776).